src/serviceWorkerRegistration.js

```diff
@@ -0,0 +1,122 @@
+const isLocalhost = Boolean(
+    window.location.hostname === 'localhost' ||
+        // [::1] is the IPv6 localhost address.
+        window.location.hostname === '[::1]' ||
+        // 127.0.0.0/8 are considered localhost for IPv4.
+        window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/)
+);
+
+export function register(config) {
+    if (process.env.NODE_ENV === 'production' && 'serviceWorker' in navigator) {
+        // The URL constructor is available in all browsers that support SW.
+        const publicUrl = new URL(process.env.PUBLIC_URL, window.location.href);
+        if (publicUrl.origin !== window.location.origin) {
+            return;
+        }
+
+        window.addEventListener('load', () => {
+            const swUrl = `${process.env.PUBLIC_URL}/service-worker.js`;
```

Comment:
Same applies here. How about using the window hostname?

Reply:
`window.location.hostname` returns the hostname of the current page. What we are trying to get from the `process.env.PUBLIC_URL` is the path from the `/build` directory to our `index.html` file. And this environment variable is injected by React and we can use it not only in the service-worker.js but anywhere else throughout the react app :)